aten/src/ATen/record_function.cpp

```diff
@@ -212,16 +212,58 @@ inline CallbackManager& manager() {
   return _manager;
 }
 
-} // namespace
+// Low probability constant
+const double kLowProb = 0.001;
+thread_local int tries_left_ = 0;
 
-/* static */
-double RecordFunctionCallback::sample_zero_one() {
+int sample_geometric() {
+  static thread_local auto gen =
+      std::make_unique<std::mt19937>(std::random_device()());
```

Comment:
if random generation is still an issue, we can experiment with minstd_rand0 or knuth_b engines - we don't need high quality randoms here.